js/reference.js

```diff
@@ -1839,7 +1841,27 @@
                 var config = {
                     headers: this._generateContextHeader(contextHeaderParams)
                 };
-                this._server._http.delete(this.location.ermrestCompactUri, config).then(function (deleteResponse) {
+
+                var uri = this.table.uri;
+
+                var keys = [];
+                //create key for each tuple and push it into the key array
+                for (var i=0; i<tuples.length; i++) {
+                    // NOTE: should probably be factored out into a function of table
+                    var keyPair = "";
+                    for (var j=0; j<this.table.shortestKey.length; j++) {
+                        var colName = this.table.shortestKey[j].name;
+                        if (!tuples[i][colName]) throw new module.BadRequestError(400, "tuples[" + i + "] does not have any data for shortest key column: " + colName);
```

Comment:
As we discussed this should be a `verify` statement, so we're sending an ermrestjs error and not ermrest. We don't need to mask this as an ermrest error.